From de4142e6c10ca025b66ec2ea934a1987c6618c29 Mon Sep 17 00:00:00 2001
From: Jordan R Abrahams-Whitehead <ajordanr@google.com>
Date: Tue, 4 Apr 2023 21:36:08 +0000
Subject: [PATCH] Allow --ex-pkg flags to append version info

At present, --ex-pkg flags always install the [latest] version.
However, sometimes one may want to install [stable] if possible.

This CL sets extra packages to use whatever the $DEFAULT_VER is set
to (which is configured via --stable/-S).
---
 crossdev | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/crossdev b/crossdev
index befce40..cd7d3c2 100755
--- a/crossdev
+++ b/crossdev
@@ -950,6 +950,11 @@ if [[ ${LPKG} == "newlib" && ${LVER} == "[stable]" ]]; then
 	LVER="[latest]"
 fi
 
+# Configure --ex-pkg entry versions, based on DEFAULT_VER.
+for _ in "${XPKGS[@]}"; do
+	XVERS+=( "${DEFAULT_VER}" )
+done
+
 show_target_cfg() {
 	local pkgs crosspkgs=()
 
-- 
2.41.0.585.gd2178a4bd4-goog

